### PR TITLE
Support xyzservices.TileProvider as an input of get_provider

### PR DESCRIPTION
**Pull request by martinfleis**
Adding support of `xyzservices.TileProvider` as an input of `get_provider` following the suggestion 1) from #11500.

- [x] issues: xref  #11500 
- [x] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by martinfleis** [comment]
@mattpap Just FYI, I haven't linked this PR to #11500 via closing keyword as this PR should not close the issue. It tackles only the first part of it.

File: bokeh/tile_providers.py [hunk 1]
```diff
@@ -232,6 +241,15 @@ def get_provider(self, provider_name):
...
             return WMTSTileSource(url=provider_name.url, attribution=provider_name.attribution)
 
+        if xyzservices:
+            if isinstance(provider_name, xyzservices.TileProvider):
+                return WMTSTileSource(
+                    url=provider_name.build_url(scale_factor="@2x"),
+                    attribution=provider_name.html_attribution,
+                    min_zoom=provider_name.get("min_zoom", 0),
+                    max_zoom=provider_name.get("max_zoom", 30),
+                )
```
Comment (bryevdv) [inline bokeh/tile_providers.py:251]: generally speaking we follow this pattern for less nesting:
```suggestion
        if xyzservices and isinstance(provider_name, xyzservices.TileProvider):
            return WMTSTileSource(
                url=provider_name.build_url(scale_factor="@2x"),
                attribution=provider_name.html_attribution,
                min_zoom=provider_name.get("min_zoom", 0),
                max_zoom=provider_name.get("max_zoom", 30),
            )
```

**Comment by bryevdv** [comment]
Pretty much LGTM just one code suggestion. It might be nice to add a new example or update one of the existing tile examples to actually demonstrate using an `xyzservices` provider.

**Comment by martinfleis** [comment]
> It might be nice to add a new example or update one of the existing tile examples to actually demonstrate using an xyzservices provider.

Will do.

**Comment by bryevdv** [comment]
We don't enforce import comment headers in examples but we require the same stdlib/ext/bk section ordering throughout:
```
       @@ -1,6 +1,7 @@
50 E         +import xyzservices.providers as xyz
51 E         +
52 E          from bokeh.plotting import figure, output_file, show
53 E          from bokeh.tile_providers import get_provider
54 E         -import xyzservices.providers as xyz
55 E          
56 E          output_file("xyzservices.html")
```

Also FYI we are in the process of dropping the `output_file` from most all the examples as it is not really needed unless you want to explicity change the filename. I would ask to just remove the `output_file` (and related import)

**Comment by bryevdv** [comment]
this LGTM added @tcmetzger in case he'd like to do a once-over the docs

BTW @martinfleis it seems like the WIkimedia tile urls no longer function? Do you know if that is the case, and if we should remove it from `tile_providers`?

**Review by bryevdv — APPROVED** [review]

**Comment by martinfleis** [comment]
> BTW @martinfleis it seems like the WIkimedia tile urls no longer function? Do you know if that is the case, and if we should remove it from tile_providers?

@bryevdv Wikimedia tiles are no longer public. They're now restricted to Wiki's own domains only, so I'd suggest removing them.

**Comment by bryevdv** [comment]
Thanks @martinfleis I've opened https://github.com/bokeh/bokeh/issues/11515

**Pull request merged**